drivers/usb/usbport/endpoint.c

```diff
@@ -1357,9 +1357,9 @@ USBPORT_InvalidateEndpointHandler(IN PDEVICE_OBJECT FdoDevice,
     {
         KeAcquireSpinLock(&FdoExtension->EndpointListSpinLock, &OldIrql);
 
-        Entry = &FdoExtension->EndpointList;
-
-        while (Entry && Entry != &FdoExtension->EndpointList)
+        for (Entry = FdoExtension->EndpointList.Flink;
+             Entry && Entry != &FdoExtension->EndpointList;
```

Comment:
Not this PR, but: is the `NULL` check actually needed?

Reply:
Yeah, it should probably be changed to an `ASSERT`

Reply:
Noted, as a future follow-up.